### PR TITLE
chore: adding eslint and prettier to the project

### PR DESCRIPTION
**Pull request by victor-heliomar**
## Overview

Added Eslint, Prettier and Husky to the project. You can check directly these commits, since prettier added too many changes:

- https://github.com/victor-heliomar/i18n-populator/commit/dd3a07dd93e7eb0d7e1acd875a971f1df597fb40
- https://github.com/victor-heliomar/i18n-populator/commit/a259de02c9c12e2517fa7c0a4e551e7181f66d7c
- https://github.com/victor-heliomar/i18n-populator/commit/6274b8b774cea493ab540c6074c821803d343657

> [!WARNING]
> Husky doesn't supports node 14 so I had to remove it from the tests. But it is ok, it would work for node 14 without problem, since husky is only a Development dependency

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Israel-Laguan) [inline .prettierrc:1]: Can this file be erased?

File: controllers/translateController.js [hunk 2]
```diff
@@ -1,34 +1,62 @@
...
   validateSettingsFile(settingsFilePath);
   validateLanguageRequested(sourceLanguage, options.engine);
-  
-  if (!text) throw new Error('No text to translate provided');
-  if (!nameOfTranslation) throw new Error('No name of translation provided');
-  
-  const { languages, basePath, translationEngines: settingsTranslationEngines } = require(settingsFilePath);
 
-  if (options.engine && !isEngineValid(options.engine)) throw new Error(`You've provided an invalid engine as arg on your CLI Command. Try with one of these: ${validEngines.join(', ')}`);
+  if (!text) throw new Error("No text to translate provided");
```
Comment (Israel-Laguan) [inline controllers/translateController.js:27]: Not sure if text should be validated different, for example the length or the type

File: cli.js [hunk 3]
```diff
@@ -1,33 +1,50 @@
...
+  generateConfigController,
+} = require("./controllers/generateConfigController");
 
 program
-  .name('i18n-populator')
-  .description('CLI to translate JavaScript strings')
-  .version('1.0.0');
+  .name("i18n-populator")
+  .description("CLI to translate JavaScript strings")
+  .version("1.0.0");
```
Comment (Israel-Laguan) [inline cli.js:15]: What about using the version from package.json?

File: .prettierrc (hunk 1 above)
Comment (victor-heliomar) [reply .prettierrc:1]: erased

File: controllers/translateController.js (hunk 2 above)
Comment (victor-heliomar) [reply controllers/translateController.js:27]: changed

File: cli.js (hunk 3 above)
Comment (victor-heliomar) [reply cli.js]: done. Thanks for the suggestion!

**Comment by victor-heliomar** [comment]
@Israel-Laguan pushed a commit solving your comments. Thanks for helping with review!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -74,8 +74,8 @@ This is a list of ideas that which you can check if you wants collaborate on the
 3. Collaborate on side projects related to the project
 
 - We are currently working on some projects related to i18n translation generator in order to improve the way you create i18n translations the most possible. Projects in mind are:
-   1. Landing Page to explain easily the idea to developers.
-   2. Documentation Page in order to improve the doc of the project and make more easier the use of it.
-   3. Visual Studio Code Extension in order to create a user interface for users that doesn't feel comfortable with CLI Commands.
+  1.  Landing Page to explain easily the idea to developers.
```
Comment (Israel-Laguan) [inline CONTRIBUTING.md:77]: NIT: It seems like there is an extra space?

File: controllers/generateConfigController.js [hunk 5]
```diff
@@ -33,7 +33,7 @@ const _promptBasePath = async () => {
   do {
     let hasError = false;
     basePath = promptUserInput(
-      'Base path for the translations files: e.g. "src/localizations.": '
+      'Base path for the translations files: e.g. "src/localizations.": ',
```
Comment (Israel-Laguan) [inline controllers/generateConfigController.js:36]: Not sure if the point at `localizations.` is needed? Or add backstick maybe

File: controllers/generateConfigController.test.js [hunk 6]
```diff
@@ -100,14 +100,14 @@ describe("_promptBasePath", () => {
...
-    mockPromptBasePath({ basePath, pathFiles })
+    mockPromptBasePath({ basePath, pathFiles });
 
     const result = await _promptBasePath();
 
     expect(promptUserInput).toHaveBeenCalledTimes(2);
     expect(promptUserInput).toHaveBeenCalledWith(
-      'Base path for the translations files: e.g. "src/localizations.": '
-      );
+      'Base path for the translations files: e.g. "src/localizations.": ',
```
Comment (Israel-Laguan) [inline controllers/generateConfigController.test.js:109]: Same here about the leading point

File: controllers/generateConfigController.test.js [hunk 7]
```diff
@@ -119,31 +119,32 @@ describe("_promptBasePath", () => {
...
 
-    mockPromptBasePath({ basePath: "test-configs/translations", pathFiles })
+    mockPromptBasePath({ basePath: "test-configs/translations", pathFiles });
 
     const result = await _promptBasePath();
 
     expect(promptUserInput).toHaveBeenCalledTimes(3);
     expect(promptUserInput).toHaveBeenCalledWith(
-      'Base path for the translations files: e.g. "src/localizations.": '
+      'Base path for the translations files: e.g. "src/localizations.": ',
```
Comment (Israel-Laguan) [inline controllers/generateConfigController.test.js:128]: Same here for the leading point

File: controllers/generateConfigController.test.js [hunk 8]
```diff
@@ -153,7 +154,7 @@ describe("_promptBasePath", () => {
...
```
Comment (Israel-Laguan) [inline controllers/generateConfigController.test.js:157]: Same here for leading point

File: controllers/generateConfigController.test.js [hunk 9]
```diff
@@ -165,25 +166,21 @@ describe("_promptBasePath", () => {
...
-      
+
     mockPromptBasePath({ basePath, pathFiles });
 
     const result = await _promptBasePath();
 
     expect(promptUserInput).toHaveBeenCalledTimes(4);
     expect(promptUserInput).toHaveBeenCalledWith(
-      'Base path for the translations files: e.g. "src/localizations.": '
+      'Base path for the translations files: e.g. "src/localizations.": ',
```
Comment (Israel-Laguan) [inline controllers/generateConfigController.test.js:183]: Same about leading point

**Review by Israel-Laguan — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (victor-heliomar) [reply CONTRIBUTING.md:77]: You're right. Removed them. Thanks!

File: controllers/generateConfigController.js (hunk 5 above)
Comment (victor-heliomar) [reply controllers/generateConfigController.js:36]: not needed hehe, you're right. thanks!

**Pull request merged**